src/components/Garden/ModalFlows/ChallengeProposalScreens/ChallengeProposalScreens.js

```diff
@@ -7,51 +7,68 @@ import { useDisputeFees } from '@hooks/useDispute'
 import { useAgreement } from '@hooks/useAgreement'
 import BigNumber from '@lib/bigNumber'
 import { toDecimals } from '@utils/math-utils'
+import { addressesEqual } from '@1hive/connect-core'
 
 const ZERO_BN = new BigNumber(toDecimals('0', 18))
 
 function ChallengeProposalScreens({ agreementActions, proposal }) {
   const [transactions, setTransactions] = useState([])
   const [agreement, loading] = useAgreement()
-  const { token } = useGardenState()
+  const { token, wrappableToken } = useGardenState()
   const disputeFees = useDisputeFees()
 
   const temporatyTrx = useRef([])
 
-  const depositAmount = useMemo(() => {
-    if (disputeFees.loading) {
-      return
-    }
-    return proposal.collateralRequirement.challengeAmount
-      .plus(new BigNumber(disputeFees.amount.toString()))
-      .toString()
-  }, [disputeFees, proposal])
-
-  const getTransactions = useCallback(
-    async (
-      onComplete,
-      actionId,
-      settlementOffer,
-      challengerFinishedEvidence,
-      context
-    ) => {
-      const allowance = await agreementActions.getAllowance()
-      if (allowance.lt(depositAmount)) {
-        if (!allowance.eq(0)) {
-          await agreementActions.approveChallengeTokenAmount(
+  const approveTokenAmount = useCallback(
+    async (tokenAddress, amount) => {
+      const tokenAllowance = await agreementActions.getAllowance(tokenAddress)
+      if (tokenAllowance.lt(amount)) {
+        if (!tokenAllowance.eq(0)) {
+          await agreementActions.approveTokenAmount(
+            tokenAddress,
             ZERO_BN,
             intent => {
               temporatyTrx.current = temporatyTrx.current.concat(intent)
             }
           )
         }
-        await agreementActions.approveChallengeTokenAmount(
-          depositAmount,
+        await agreementActions.approveTokenAmount(
+          tokenAddress,
+          amount,
           intent => {
             temporatyTrx.current = temporatyTrx.current.concat(intent)
           }
         )
       }
+    },
+    [agreementActions]
+  )
+
+  const getTransactions = useCallback(
+    async (
+      onComplete,
+      actionId,
+      settlementOffer,
+      challengerFinishedEvidence,
+      context
+    ) => {
+      const collateralToken = proposal.collateralRequirement.tokenId
+      const collateralAmount = proposal.collateralRequirement.challengeAmount
```

Comment:
```suggestion
      const {challengeAmount: collateralAmount, tokenId: collateralToken} = proposal.collateralRequirement
```

Reply:
In this case i would prefer to keep is as since the suggested code takes 4 lines when auto-linted